### ISSUE TITLE
Does not builds for ruby 2.3.0

### ISSUE THREAD

**Issue by zuf** [ruby-opencv/ruby-opencv#75]
ruby 2.3.0
gcc 5.3.0
opencv 2.4.12.2

With ruby 2.1.10 installed just fine.

```
$ gem install ruby-opencv 

Building native extensions.  This could take a while...
ERROR:  Error installing ruby-opencv:
	ERROR: Failed to build gem native extension.

    current directory: /home/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/ruby-opencv-0.0.14/ext/opencv
/home/user/.rbenv/versions/2.3.0/bin/ruby -r ./siteconf20160407-11452-15har8q.rb extconf.rb
>> Check the required libraries...
checking for main() in -lstdc++... yes
checking for main() in -lopencv_calib3d... yes
checking for main() in -lopencv_contrib... yes
checking for main() in -lopencv_core... yes
checking for main() in -lopencv_features2d... yes
checking for main() in -lopencv_flann... yes
checking for main() in -lopencv_highgui... yes
checking for main() in -lopencv_imgproc... yes
checking for main() in -lopencv_legacy... yes
checking for main() in -lopencv_ml... yes
checking for main() in -lopencv_objdetect... yes
checking for main() in -lopencv_video... yes
checking for main() in -lopencv_photo... yes
checking for main() in -lopencv_gpu... yes
checking for main() in -lopencv_nonfree... yes
>> Check the required headers...
checking for opencv2/core/core_c.h... yes
checking for opencv2/core/core.hpp... yes
checking for opencv2/imgproc/imgproc_c.h... yes
checking for opencv2/imgproc/imgproc.hpp... yes
checking for opencv2/video/tracking.hpp... yes
checking for opencv2/features2d/features2d.hpp... yes
checking for opencv2/flann/flann.hpp... yes
checking for opencv2/calib3d/calib3d.hpp... yes
checking for opencv2/objdetect/objdetect.hpp... yes
checking for opencv2/legacy/compat.hpp... yes
checking for opencv2/legacy/legacy.hpp... yes
checking for opencv2/highgui/highgui_c.h... yes
checking for opencv2/highgui/highgui.hpp... yes
checking for opencv2/photo/photo.hpp... yes
checking for opencv2/nonfree/nonfree.hpp... yes
checking for stdarg.h... yes
creating Makefile

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/extensions/x86_64-linux/2.3.0-static/ruby-opencv-0.0.14/mkmf.log

current directory: /home/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/ruby-opencv-0.0.14/ext/opencv
make "DESTDIR=" clean

current directory: /home/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/ruby-opencv-0.0.14/ext/opencv
make "DESTDIR="
compiling cvpoint.cpp
compiling lbph.cpp
compiling iplimage.cpp
compiling algorithm.cpp
In file included from /home/user/.rbenv/versions/2.3.0/include/ruby-2.3.0/ruby.h:33:0,
                 from opencv.h:17,
                 from algorithm.h:13,
                 from algorithm.cpp:11:
algorithm.cpp: In function 'VALUE mOpenCV::cAlgorithm::rb_set_matvector(VALUE, VALUE, VALUE)':
/home/user/.rbenv/versions/2.3.0/include/ruby-2.3.0/ruby/ruby.h:882:52: error: '__builtin_choose_expr' was not declared in this scope
      OBJ_WB_UNPROTECT((VALUE)(obj)), ((VALUE)(obj))))
                                                    ^
/home/user/.rbenv/versions/2.3.0/include/ruby-2.3.0/ruby/ruby.h:1039:48: note: in definition of macro 'RARRAY_CONST_PTR'
 #define RARRAY_CONST_PTR(a) rb_array_const_ptr(a)
                                                ^
/home/user/.rbenv/versions/2.3.0/include/ruby-2.3.0/ruby/ruby.h:1059:50: note: in expansion of macro 'RB_OBJ_WB_UNPROTECT_FOR'
 #define RARRAY_PTR(a) ((VALUE *)RARRAY_CONST_PTR(RB_OBJ_WB_UNPROTECT_FOR(ARRAY, a)))
                                                  ^
algorithm.cpp:108:24: note: in expansion of macro 'RARRAY_PTR'
     VALUE* value_ptr = RARRAY_PTR(value);
                        ^
Makefile:230: recipe for target 'algorithm.o' failed
make: *** [algorithm.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/ruby-opencv-0.0.14 for inspection.
Results logged to /home/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/extensions/x86_64-linux/2.3.0-static/ruby-opencv-0.0.14/gem_make.out
```

**Comment by ser1zw** [comment]
Hi,

It seems to be the same problem as https://bugs.ruby-lang.org/issues/11962 .
I'll implement a workaround.

**Comment by ser1zw** [comment]
Fixed in v0.0.15 https://rubygems.org/gems/ruby-opencv/versions/0.0.15
